mbed-hal/serial_api.h

```diff
@@ -175,6 +186,16 @@ void serial_putc(serial_t *obj, int c);
  */
 int  serial_readable(serial_t *obj);
 
+/** Check if any errors were encountered by the serial peripheral:
+ *  RX overrun, parity error, framing error, or FIFO overflow. The
+ *  value is a bit field where one bit represents one error type
+ * (in the mentioned order starting from the least significant bit).
+ *
+ * @param obj The serial object
+ * @return Non-zero value if any of the errors have occurred, 0 otherwise
```

Comment:
Updated the documentation. Added a few @retval tags as an example; didn't want to enumerate the whole list of values. A developer should get the idea...